### ISSUE TITLE
"vulnerable" should not be green

### ISSUE THREAD

**Issue by jonasstein** [repology/repology-webapp#156]
green should be used for good/safe situations.

https://repology.org/projects/?maintainer=desktop-misc%40gentoo.org&inrepo=gentoo&problematic=1
![2020-11-12_screenshot_0c8fda3e](https://user-images.githubusercontent.com/298682/98991180-ca225080-252b-11eb-8e9d-a381f9dfb0df.png)

vulnerable▲ - version is potentially vulnerable as there are related CVEs.

I suggest white on red.

**Comment by AMDmi3** [comment]
I take it as you've misunderstood the legend. Vulnerable status is denoted by the yellow warning sign, not the background color. The color indicates version status (up to date/outdated).

**Comment by KOLANICH** [comment]
>Vulnerable status is denoted by the yellow warning sign

Why not a scull and bones emoji?

**Comment by jonasstein** [comment]
@AMDmi3 OK, I understand. However the impression is misleading, because the green dominates and the yellow triangle is so tiny. I still suggest to change it to a more significant symbol like a large UTF8 ⚠, or

![2021-10-04_screenshot_3f7853a4](https://user-images.githubusercontent.com/298682/135922135-3cf1e1b8-bb72-4894-920c-2c84ac277b17.png)

instead of

![2021-10-04_screenshot_35dec796](https://user-images.githubusercontent.com/298682/135921715-8599c5f7-b799-41ca-a4ad-e5181de0aa49.png)

**Comment by AMDmi3** [comment]
This looks reasonable.

**Comment by FedericoCeratto** [comment]
The background color is the most visually striking element. It can appear alarming on badges.
I suggest using a red background exclusively for vulnerable packages and not for "outdated" versions.

**Comment by AMDmi3** [comment]
> I suggest using a red background exclusively for vulnerable packages and not for "outdated" versions.

Definitely no. Vulnerabilities are not the main topic of Repology, and also NVD quality and coverage is lacking, making it not suitable for primary package classification. However, vulnerabilities correlate pretty well with software freshness, and in most cases when new CVE is published, (only) the latest software version is not susceptible to it. So even for this purpose, it's still sensible to rely solely on freshness, and that also works for all cases not covered by NVD, such as not (yet) known vulnerabilities, not published vulnerabilities, vulnerabilities not yet analyzed and properly filled in NVD, CVES not matched with projects and CVES with incorrect versions not suitable for matching.

**Comment by FedericoCeratto** [comment]
> vulnerabilities correlate pretty well with software freshness

This is not true in general. Security-oriented distributions develop and distribute security updates for stable and LTS releases of the distribution.
Over time, **if** more vulnerabilities are discovered, published and fixed the level of security of "legacy" software improves.
On the other hand, feature releases from upstream might provide security fixes but also contain newly created vulnerabilities.
(At risk of stating the obvious: vulnerabilities can be discovered at any time but can only be created by new releases)
